sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTest.groovy

```diff
@@ -0,0 +1,693 @@
+import com.azure.storage.blob.models.BlobErrorCode
+import com.azure.storage.blob.models.BlobImmutabilityPolicy
+import com.azure.storage.blob.models.BlobImmutabilityPolicyMode
+import com.azure.storage.blob.models.BlobListDetails
+import com.azure.storage.blob.models.BlobRequestConditions
+import com.azure.storage.blob.models.BlobStorageException
+import com.azure.storage.blob.models.LeaseStateType
+import com.azure.storage.blob.models.ListBlobContainersOptions
+import com.azure.storage.blob.models.ListBlobsOptions
+import com.azure.storage.blob.models.ParallelTransferOptions
+import com.azure.storage.blob.models.PublicAccessType
+import com.azure.storage.blob.options.AppendBlobCreateOptions
+import com.azure.storage.blob.options.BlobBeginCopyOptions
+import com.azure.storage.blob.options.BlobBreakLeaseOptions
+import com.azure.storage.blob.options.BlobCopyFromUrlOptions
+import com.azure.storage.blob.options.BlobParallelUploadOptions
+import com.azure.storage.blob.options.BlockBlobCommitBlockListOptions
+import com.azure.storage.blob.options.BlockBlobSimpleUploadOptions
+import com.azure.storage.blob.options.PageBlobCreateOptions
+import com.azure.storage.blob.sas.BlobContainerSasPermission
+import com.azure.storage.blob.sas.BlobSasPermission
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues
+import com.azure.storage.common.implementation.Constants
+import com.azure.storage.common.sas.AccountSasPermission
+import com.azure.storage.common.sas.AccountSasResourceType
+import com.azure.storage.common.sas.AccountSasService
+import com.azure.storage.common.sas.AccountSasSignatureValues
+import com.azure.storage.common.test.shared.extensions.LiveOnly
+import com.azure.storage.common.test.shared.extensions.RequiredServiceVersion
+import com.fasterxml.jackson.databind.ObjectMapper
+import reactor.core.publisher.Flux
+import reactor.netty.http.client.HttpClient
+import spock.lang.ResourceLock
+import spock.lang.Unroll
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.time.Duration
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+
+@ResourceLock("ManagementPlaneThrottling")
+class ImmutableStorageWithVersioningTest extends APISpec {
+
+    private BlobContainerClient vlwContainer;
+    private BlobClient vlwBlob
+    private String accountName = env.primaryAccount.name
+    private String containerName
+    private String resourceGroupName = "XClient"
+    private String subscriptionId = "ba45b233-e2ef-4169-8808-49eb0d8eba0d"
+    private String apiVersion = "2021-04-01"
+    private TokenCredential credential = new EnvironmentCredentialBuilder().build()
+    private BearerTokenAuthenticationPolicy credentialPolicy = new BearerTokenAuthenticationPolicy(credential, "https://management.azure.com/.default")
+
+    def setup() {
+        containerName = generateContainerName()
+
+        if (env.testMode != TestMode.PLAYBACK) {
```

Comment:
Ignore some of the ugliness here in setup and cleanup. This is temporary until the Management SDK for Java releases support.

Reply:
Have we captured this in a form of GH issue?

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/22368